Accessing empty pod list

**What happened?**:
While trying to deploy Pachyderm on Kubernetes, the pods fail to start due to a code exception

```
2020-07-21T07:58:18Z INFO no Jaeger collector found (JAEGER_COLLECTOR_SERVICE_HOST not set)
2020-07-21T07:58:18Z INFO no Jaeger collector found (JAEGER_COLLECTOR_SERVICE_HOST not set)
2020-07-21T07:58:18Z WARNING TLS disabled: could not stat public cert at /pachd-tls-cert/tls.crt: stat /pachd-tls-cert/tls.crt: no such file or directory
2020-07-21T07:58:18Z INFO started setting up External Pachd GRPC Server
2020-07-21T07:58:18Z INFO started setting up PFS API GRPC Server
2020-07-21T07:58:18Z INFO finished setting up PFS API GRPC Server
2020-07-21T07:58:18Z INFO started setting up PPS API GRPC Server
2020-07-21T07:58:18Z ERROR able to access kubernetes pods, but did not find a pachd pod... this is very strange since this code is run from within a pachd pod
2020-07-21T07:58:18Z INFO finished setting up PPS API GRPC Server
2020-07-21T07:58:18Z INFO finished setting up External Pachd GRPC Server
panic: runtime error: index out of range [0] with length 0
goroutine 1 [running]:
github.com/pachyderm/pachyderm/src/server/pps/server.(*apiServer).validateKube(0xc0000ef8c0)
	/app/src/server/pps/server/api_server.go:326 +0xb50
github.com/pachyderm/pachyderm/src/server/pps/server.NewAPIServer(0xc00012a000, 0xc0003a8320, 0xc0006e26a0, 0x1d, 0x25f3f4a, 0x7, 0xc00005609d, 0x17, 0xc000056015, 0x16, ...)
	/app/src/server/pps/server/server.go:68 +0x3d1
main.doFullMode.func3.2(0xc0006417b0, 0x2c4c8cc)
	/app/src/server/cmd/pachd/main.go:382 +0x34d
main.logGRPCServerSetup(0x2c07f52, 0x7, 0xc000641ab8, 0x0, 0x0)
	/app/src/server/cmd/pachd/main.go:776 +0x114
main.doFullMode.func3(0xc000641b40, 0x2c4c8cc)
	/app/src/server/cmd/pachd/main.go:381 +0x2f8
main.logGRPCServerSetup(0x2c153f1, 0xe, 0xc000641da8, 0x0, 0x0)
	/app/src/server/cmd/pachd/main.go:776 +0x114
main.doFullMode(0x2626340, 0xc000152780, 0x0, 0x0)
	/app/src/server/cmd/pachd/main.go:363 +0xa6d
github.com/pachyderm/pachyderm/src/server/pkg/cmdutil.Main(0x2d11ef0, 0x2626340, 0xc000152780, 0x0, 0x0, 0x0)
	/app/src/server/pkg/cmdutil/env.go:33 +0x83
main.main()
	/app/src/server/cmd/pachd/main.go:84 +0x1a9
```

**What you expected to happen?**:
The installation would be successful.

**How to reproduce it (as minimally and precisely as possible)?**:
Deploying the pachyderm version `1.11.0` to an EKS cluster version `1.15`. We use the following deployment file

```yaml
apiVersion: apps/v1beta1
kind: Deployment
metadata:
  labels:
    app: pachd
    suite: pachyderm
  name: pachd
  namespace: pachyderm
spec:
  replicas: 3
  selector:
    matchLabels:
      app: pachd
      suite: pachyderm
  template:
    metadata:
      labels:
        app: pachd
        suite: pachyderm
      name: pachd
      namespace: pachyderm
    spec:
      containers:
        - env:
            - name: PACH_ROOT
              value: /pach
            - name: NUM_SHARDS
              value: "16"
            - name: STORAGE_BACKEND
              value: MINIO
            - name: WORKER_IMAGE
              value: pachyderm/worker:1.11.0
            - name: IMAGE_PULL_SECRET
            - name: WORKER_SIDECAR_IMAGE
              value: pachyderm/pachd:1.11.0
            - name: WORKER_IMAGE_PULL_POLICY
              value: IfNotPresent
            - name: STORAGE_UPLOAD_CONCURRENCY_LIMIT
              value: "100"
            - name: PACHD_VERSION
              value: "1.11.0"
            - name: METRICS
              value: "true"
            - name: LOG_LEVEL
              value: debug
            - name: BLOCK_CACHE_BYTES
              value: 1G
            - name: NO_EXPOSE_DOCKER_SOCKET
              value: "false"
            - name: PACHYDERM_AUTHENTICATION_DISABLED_FOR_TESTING
              value: "false"
            - name: PACHD_POD_NAMESPACE
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: metadata.namespace
            - name: PACHD_MEMORY_REQUEST
              valueFrom:
                resourceFieldRef:
                  containerName: pachd
                  divisor: "0"
                  resource: requests.memory
            - name: EXPOSE_OBJECT_API
              value: "false"
            - name: MICROSOFT_SECRET
              valueFrom:
                secretKeyRef:
                  key: microsoft-secret
                  name: pachyderm-storage-secret
                  optional: true
            - name: MINIO_ENDPOINT
              valueFrom:
                secretKeyRef:
                  key: minio-endpoint
                  name: pachyderm-storage-secret
                  optional: true
            - name: MINIO_ID
              valueFrom:
                secretKeyRef:
                  key: minio-id
                  name: pachyderm-storage-secret
                  optional: true
            - name: AMAZON_REGION
              valueFrom:
                secretKeyRef:
                  key: amazon-region
                  name: pachyderm-storage-secret
                  optional: true
            - name: AMAZON_SECRET
              valueFrom:
                secretKeyRef:
                  key: amazon-secret
                  name: pachyderm-storage-secret
                  optional: true
            - name: AMAZON_DISTRIBUTION
              valueFrom:
                secretKeyRef:
                  key: amazon-distribution
                  name: pachyderm-storage-secret
                  optional: true
            - name: GOOGLE_BUCKET
              valueFrom:
                secretKeyRef:
                  key: google-bucket
                  name: pachyderm-storage-secret
                  optional: true
            - name: GOOGLE_CRED
              valueFrom:
                secretKeyRef:
                  key: google-cred
                  name: pachyderm-storage-secret
                  optional: true
            - name: MINIO_BUCKET
              valueFrom:
                secretKeyRef:
                  key: minio-bucket
                  name: pachyderm-storage-secret
                  optional: true
            - name: MINIO_SIGNATURE
              valueFrom:
                secretKeyRef:
                  key: minio-signature
                  name: pachyderm-storage-secret
                  optional: true
            - name: MINIO_SECURE
              valueFrom:
                secretKeyRef:
                  key: minio-secure
                  name: pachyderm-storage-secret
                  optional: true
            - name: AMAZON_ID
              valueFrom:
                secretKeyRef:
                  key: amazon-id
                  name: pachyderm-storage-secret
                  optional: true
            - name: AMAZON_TOKEN
              valueFrom:
                secretKeyRef:
                  key: amazon-token
                  name: pachyderm-storage-secret
                  optional: true
            - name: AMAZON_VAULT_ADDR
              valueFrom:
                secretKeyRef:
                  key: amazon-vault-addr
                  name: pachyderm-storage-secret
                  optional: true
            - name: AMAZON_VAULT_ROLE
              valueFrom:
                secretKeyRef:
                  key: amazon-vault-role
                  name: pachyderm-storage-secret
                  optional: true
            - name: AMAZON_VAULT_TOKEN
              valueFrom:
                secretKeyRef:
                  key: amazon-vault-token
                  name: pachyderm-storage-secret
                  optional: true
            - name: MICROSOFT_CONTAINER
              valueFrom:
                secretKeyRef:
                  key: microsoft-container
                  name: pachyderm-storage-secret
                  optional: true
            - name: MICROSOFT_ID
              valueFrom:
                secretKeyRef:
                  key: microsoft-id
                  name: pachyderm-storage-secret
                  optional: true
            - name: MINIO_SECRET
              valueFrom:
                secretKeyRef:
                  key: minio-secret
                  name: pachyderm-storage-secret
                  optional: true
            - name: AMAZON_BUCKET
              valueFrom:
                secretKeyRef:
                  key: amazon-bucket
                  name: pachyderm-storage-secret
                  optional: true
          image: pachyderm/pachd:1.11.0
          imagePullPolicy: IfNotPresent
          name: pachd
          ports:
            - containerPort: 650
              name: api-grpc-port
              protocol: TCP
            - containerPort: 651
              name: trace-port
            - containerPort: 652
              name: api-http-port
              protocol: TCP
            - containerPort: 653
              name: peer-port
              protocol: TCP
            - containerPort: 999
              name: api-git-port
              protocol: TCP
            - containerPort: 654
              name: saml-port
              protocol: TCP
          readinessProbe:
            exec:
              command:
                - /pachd
                - --readiness
          resources:
            limits:
              cpu: "1"
              memory: 3G
            requests:
              cpu: "1"
              memory: 3G
          volumeMounts:
            - mountPath: /pach
              name: pach-disk
            - mountPath: /pachyderm-storage-secret
              name: pachyderm-storage-secret
      serviceAccountName: pachyderm
      volumes:
        - name: pach-disk
        - name: pachyderm-storage-secret
          secret:
            secretName: pachyderm-storage-secret
```

**Anything else we need to know?**:
Analyzing the code base, I noticed that [here](https://github.com/pachyderm/pachyderm/blob/93d3fc80333a4f29d1a9d18731435c909e041e02/src/server/pps/server/api_server.go#L326) you do a check if the `len(pods) == 0`. If the list is empty (and I believe `nil` as well), you log the error. However, the code continues and it tries to access the first element of an `empty/nil` list. I feel it misses also `errors = true`.
Or should it break beforehand? :smile:

**Environment?**:
- Kubernetes version (use `kubectl version`): **1.15**
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s): **AWS EKS**

--- a/src/server/pps/server/api_server.go
+++ b/src/server/pps/server/api_server.go
@@ -315,13 +315,10 @@ func (a *apiServer) validateKube() {
 		logrus.Errorf("unable to access kubernetes pods, Pachyderm will continue to work but certain pipeline errors will result in pipelines being stuck indefinitely in \"starting\" state. error: %v", err)
 	}
 	pods, err := a.rcPods("pachd")
-	if err != nil {
+	if err != nil || len(pods) == 0 {
 		errors = true
 		logrus.Errorf("unable to access kubernetes pods, Pachyderm will continue to work but 'pachctl logs' will not work. error: %v", err)
 	} else {
-		if len(pods) == 0 {
-			logrus.Errorf("able to access kubernetes pods, but did not find a pachd pod... this is very strange since this code is run from within a pachd pod")
-		}
 		// No need to check all pods since we're just checking permissions.
 		pod := pods[0]
 		_, err = kubeClient.CoreV1().Pods(a.namespace).GetLogs(

